dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateFactoryImpl.java

```diff
@@ -78,16 +72,20 @@ public Template find(String inode) throws DotStateException, DotDataException {
 		return template;
 	}
 
-	
-	
-	
 	@SuppressWarnings("unchecked")
-	public List<Template> findTemplatesAssignedTo(Host parentHost, boolean includeArchived) throws DotHibernateException {
-		HibernateUtil hu = new HibernateUtil(Template.class);
-		String query = !includeArchived?templatesUnderHostSQL + " and vi.deleted = " + DbConnectionFactory.getDBFalse():templatesUnderHostSQL;
-		hu.setSQLQuery(query);
-		hu.setParam(parentHost.getIdentifier());
-		return new ArrayList<Template>(new HashSet<Template>(hu.list()));
+	public List<Template> findTemplatesAssignedTo(Host parentHost, boolean includeArchived)
+			throws DotDataException {
+		DotConnect dc = new DotConnect();
+		String query = !includeArchived ? templatesUnderHostSQL + " and vi.deleted = "
+				+ DbConnectionFactory.getDBFalse() : templatesUnderHostSQL;
+		dc.setSQL(query);
+		dc.addParam(parentHost.getIdentifier());
+
+		try {
+			return ConvertToPOJOUtil.convertDotConnectMapToTemplate(dc.loadResults());
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)